#!/usr/bin/env python
PACKAGE = "franka_interactive_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("damping_eigval0", double_t,    0,     "damping eigenvalue 0", 0,  0, 300)
gen.add("damping_eigval1", double_t,    0,     "damping eigenvalue 1", 0,  0, 300)
# gen.add("rot_stiffness",   double_t,    0,     "set rotation error stiffness", 0,  0, 50)
# gen.add("rot_damping",     double_t,    0.05,  "set rotation error stiffness", 0.05,  0, 0.5)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 30, 0, 50)
gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 0.1, 0, 100)
gen.add("update_impedance_params", bool_t, 0, "Will update the impedance parameters when set to True", False)


gen.add("activate_tool_compensation", bool_t, 0, "trigger to activate external tool compensation (the desired compensation forces defined in ./config)", True)
gen.add("activate_angular_passiveDS", bool_t, 0, "trigger to activate external tool compensation (the desired compensation forces defined in ./config)", True)


exit(gen.generate(PACKAGE, "dynamic_passive_ds", "passive_ds_param"))